crates/lang/src/traits.rs

```diff
@@ -120,3 +121,63 @@ pub trait MessageMut: FnInput + FnOutput + FnSelector + FnState {
 /// Indicates that some compile time expression is expected to be `true`.
 #[doc(hidden)]
 pub trait True {}
+
+/// The ECDSA compressed public key.
+#[derive(Debug, Copy, Clone)]
+pub struct ECDSAPublicKey(pub [u8; 33]);
+
+impl Default for ECDSAPublicKey {
+    fn default() -> Self {
+        Self { 0: [0; 33] }
+    }
+}
+
+impl core::ops::Deref for ECDSAPublicKey {
+    type Target = [u8; 33];
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl core::ops::DerefMut for ECDSAPublicKey {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// Address of Ethereum account
+pub type EthereumAddress = [u8; 20];
+
+impl ECDSAPublicKey {
+    pub fn to_eth_address(&self) -> EthereumAddress {
```

Comment:
Regarding `EthereumAddress` and `to_eth_address` ‒ could you create a new crate `ink_eth_compatibility` under `crates` and migrate the struct and conversion of `ECDSAPublicKey` to `EthereumAddress` there? We might add more Ethereum-compatibility features in the future and it will be good to have a clear separation from the beginning.